### PR TITLE
dev: update starknet-foundry section to 2.7.0

### LINKED ISSUE #864: dev: update starknet-foundry section to 2.7.0
The compiler generated `<storage_var_name>ContractMemberStateTrait` traits, where `storage_var_name` is a name of a property of the `Storage` struct, no longer exist. Adapt the example so that it compiles with 2.7.0

### PR DESCRIPTION
**Pull request by TAdev0**
Closes #864

- - -
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/904)

### CODE REVIEW

**Comment by TAdev0** [comment]
@enitrat  not finished

**Comment by TAdev0** [comment]
@enitrat  ready for review

File: listings/ch17-starknet-smart-contracts-security/listing_02_pizza_factory_snfoundry/.tool-versions [hunk 1]
```diff
@@ -1 +1,2 @@
-scarb 2.6.4
+scarb 2.7.0
+starknet-foundry 0.27.0
```
Comment (enitrat) [inline listings/ch17-starknet-smart-contracts-security/listing_02_pizza_factory_snfoundry/.tool-versions:2]: unless this package requires a specific version of starknet foundry / scarb, we can remove the .tool-versions file and rely on the default ones

File: listings/ch17-starknet-smart-contracts-security/listing_02_pizza_factory_snfoundry/src/tests/foundry_test.cairo [hunk 2]
```diff
@@ -106,7 +107,8 @@ fn test_set_as_new_owner_direct() {
     let mut state = PizzaFactory::contract_state_for_testing();
     let owner: ContractAddress = contract_address_const::<'owner'>();
     state.set_owner(owner);
-    assert_eq!(state.owner.read(), owner);
+    let state_owner = state.get_owner();
```
Comment (enitrat) [inline listings/ch17-starknet-smart-contracts-security/listing_02_pizza_factory_snfoundry/src/tests/foundry_test.cairo:110]: we don't want to use `get_owner`, we want to access the `owner` storage variable directly

**Review by enitrat — APPROVED** [review]

**Comment by enitrat** [comment]
```
#[starknet::contract]
mod Hello {
    #[storage]
    pub struct Storage {
        pub value: u32
    }
}


#[cfg(test)]
mod tests {
    use core::starknet::storage::StoragePointerWriteAccess;
    use core::starknet::storage::StoragePointerReadAccess;
    use super::Hello::unsafe_new_contract_state;

    #[test]
    fn test_foo() {
        let mut state = unsafe_new_contract_state();

        let value = state.value.read();
        println!("value {}", value);

        state.value.write(42);

        let new_value = state.value.read();
        println!("new_value {}", new_value);
    }
}
```

**Comment by TAdev0** [comment]
@enitrat  i kept `contract_state_for_testing` usage, do you want me to use `unsafe_new_contract_state` like in your example? both work

**Review by enitrat — DISMISSED** [review]

**Review by enitrat — APPROVED** [review]

**Pull request merged**